src/jit/compiler.cpp

```diff
@@ -718,13 +712,25 @@ var_types  Compiler::getArgTypeForStruct(CORINFO_CLASS_HANDLE clsHnd,
         useType = GetEightByteType(structDesc, 0);
     }
 
-#else // not UNIX_AMD64
+#elif defined(_TARGET_X86_)
 
-    // We set the "primitive" useType based upon the structSize
-    // and also examine the clsHnd to see if it is an HFA of count one
-    useType = getPrimitiveTypeForStruct(structSize, clsHnd);
+    // On x86 we pass all structs by value 
+    useType = TYP_UNKNOWN;
```

Comment:
This is confusing because it contradicts the comment above that says that `useType` is `TYP_STRUCT` always if it is passed by value. However, I see that in this "phase" you are determining whether to use a primitive type, so `useType == TYP_UNKNOWN` just means that it is NOT a primitive type. I would suggest modifying the comment to `// On x64 we never pass structs as primitive types, so for now we set useType to TYP_UNKNOWN`